packages/eslint-plugin-pf-codemods/lib/rules/refactor-breakpointmods.js

```diff
@@ -0,0 +1,129 @@
+  direction: ['column', 'columnReverse', 'row', 'rowReverse'],
+  alignItems: ['alignItemsFlexStart', 'alignItemsFlexEnd', 'alignItemsCenter', 'alignItemsStretch', 'alignItemsBaseline'],
+  alignContent: ['alignContentFlexStart', 'alignContentFlexEnd', 'alignContentCenter', 'alignContentStretch', 'alignContentSpaceBetween', 'alignContentSpaceAround'],
+  alignSelf: ['alignSelfFlexStart', 'alignSelfFlexEnd', 'alignSelfCenter', 'alignSelfStretch', 'alignSelfBaseline'],
+  align: ['alignLeft', 'alignRight'],
+  justifyContent: ['justifyContentFlexStart', 'justifyContentFlexEnd', 'justifyContentCenter', 'justifyContentSpaceBetween', 'justifyContentSpaceAround', 'justifyContentSpaceEvenly'],
+  display: ['inlineFlex'],
+  fullWidth: ['fullWidth'],
+  flexWrap: ['wrap', 'wrapReverse', 'nowrap']
+};
+const errorMessages = {
+  Flex: 'Removed breakpointMods prop from Flex in favor of spacer, spaceItems, grow, shrink, flex, direction, alignItems, alignContent, alignSelf, align, justifyContent, display, fullWidth and flexWrap',
+  FlexItem: 'Removed breakpointMods prop from FlexItem in favor of spacer, grow, shrink, flex, alignSelf, align, and fullWidth',
+  ToolbarItem: 'Removed breakpointMods prop from ToolbarItem in favor of visiblity, alignment, and spacer',
+  ToolbarToggleGroup: 'Removed breakpointMods prop from ToolbarToggleGroup in favor of visiblity, alignment, spacer, and spaceItems'
+};
+
+// https://github.com/patternfly/patternfly-react/pull/4310
+module.exports = {
+  create: function(context) {
+    
+    const imports = getPackageImports(context, '@patternfly/react-core')
+      .filter(specifier => componentsToUpdate.includes(specifier.imported.name));
+      
+    return imports.length == 0 ? {} : {
+      JSXOpeningElement(node) {
+        const nodeName = node.name.name;
+        const nodeImport = imports.find(imp => imp.local.name === nodeName); 
+        if (nodeImport) {
+          const importName = nodeImport.imported.name;
+          // PageSection
+          if (importName === 'PageSection') {
+            const attribute = node.attributes.find(node => node.name && node.name.name === 'hasNoPadding');
+            if (attribute) {
+              context.report({
+                node,
+                message: `hasNoPadding prop on PageSection component removed in favor of padding={{ default: 'noPadding' }}`,
+                fix(fixer) {
+                  return fixer.replaceText(attribute, `padding={{ default: 'noPadding' }}`);
+                }
+              });
+            }
+          } else {
+            // ToolbarToggleGroup
+            if (importName === 'ToolbarToggleGroup') {
+              const attribute = node.attributes.find(attr => attr.name && attr.name.name === 'breakpoint');
+              if (attribute) {
+                const breakpointVal = attribute.value.value;
+                const newAttr = `show={{ ${breakpointVal}: 'show' }}`
```

Comment:
is this logically correct? Looking at the old code, if `breakpoint="md"` it seems that it was supposed to collapse/hide at that breakpoint.

Reply:
@jschuler thanks for calling this out - I believe the description of the previous prop was a bit confusing, it listed the `breakpoint` prop as `The breakpoint at which the toggle group is collapsed down` but is actually used as the breakpoint at which the toggle group is expanded (anything below it is collapsed) as seen in the prior usage:
`styles.modifiers[`showOn${capitalize(breakpoint)}` as 'showOnMd' | 'showOnLg' | 'showOnXl'],`

I believe the usage is correct as is in the PR, and the verbiage of `show` is a little more accurate as it's the breakpoint at which the ToolbarToggleGroup is shown in place of just the toggle icon.

Reply:
@redallen does this match with your understanding when implementing the new API?

Reply:
Yep! That's correct @evwilkin . I can see why the confusion arose...

Reply:
Thanks for checking @evwilkin !